xdebug_stack.c

```diff
@@ -1383,6 +1402,19 @@ function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_a
 	return tmp;
 }
 
+void xdebug_remove_stack_frame(function_stack_entry *fse) {
+	size_t i = 0;
+	/* Unlocks the garbage collection of stack frame (function call arguments) objects
+	 * So these objects can be properly GCed by PHP later on */
+	for (i = 0; i < fse->gc_locked_objects_count; i++) {
+		#if PHP_VERSION_ID >= 70300
+			GC_DELREF(fse->gc_locked_objects[i]);
+		#else
+			GC_REFCOUNT(fse->gc_locked_objects[i])--;
```

Comment:
I don't believe that using `GC_DELREF`/`GC_REFCOUNT` would immediately kick off the garbage collector. Would this be a problem?

Reply:
No, but we must remove a GC counter so that PHP will be able to GC "sometime". When, well, it's in PHP's end IMO so it's no Xdebug's job to trigger the GC.
I've done so because before (without removing the GC count), I managed to make a test that fails with a memory overflow (memory limit reached, I don't remember the exact PHP error message but it's the usual one). So now, this test passes. So when is GC triggers seems not a problem